### ISSUE TITLE
GetHeaders() and SetEmailSenderAndRepresentingSender() not reached.

### ISSUE THREAD

**Issue by mdocter** [Sicos1977/MSGReader#26]
Hi Kees,

My bulk import went very well thanks to your great library. I encountered only two issues.

[1] In some cases the `return` statement in the following line

[https://github.com/Sicos1977/MSGReader/blob/master/MsgReader/Outlook/Message.cs#L1279](https://github.com/Sicos1977/MSGReader/blob/master/MsgReader/Outlook/Message.cs#L1279)

Prevents the code to reach the lines at 1295 and 1296:

```C#
GetHeaders();
SetEmailSenderAndRepresentingSender();
```

That caused me to end up with empty Sender fields. I worked around this issue by moving the two method calls to lines 1225 and 1226, so before the `if` statement.
I can't send you an example msg because of customer confidentiality. I know that sucks.

[2] In some other rare case the e-mail addresses contained the Exchange Legacy Distinguished Name, so something like `/o=SomeDomain/ou=SomeGroup/cn=Recipients/cn=Morgan Cheng` instead of an e-mail address. 

In those cases I could work around most of them by running the returned e-mail addresses through the following quick hack AD lookup method (found somewhere on Stack Overflow ...I think)...
```C#
using System.DirectoryServices;

public string ResolveEmailAddress(string email)
{
	// Check if the address is set and if it's an Exchange Distinguished Name (i.e. Canonical CN=... etc)
	if (string.IsNullOrWhiteSpace(email) || !email.Contains("CN=")) return email;

	using (var ds = new DirectorySearcher())
	{
		ds.Filter = $"(&(ObjectClass=User)(legacyExchangeDN={email}))";
		var src = ds.FindAll();

		// Return the Exchange Distinguished Name in case we can't get the e-mail address for some reason
		if (src.Count != 1 || src[0] == null || src[0].Properties == null || src[0].Properties.Count == 0 || 
			!src[0].Properties.Contains("Mail") || src[0].Properties["Mail"].Count == 0) return email;
		
		return string.IsNullOrWhiteSpace(src[0].Properties["Mail"][0].ToString())
			? email
			: src[0].Properties["Mail"][0].ToString();
	}
}
```

**Comment by Sicos1977** [comment]
I did fix the first point in your issue since it doesn't matter if the 2 code lines are all the way up in the function. I did not add the second point because this assumes that the email that is being processed is originated from the same active directory domain as where the msgreader.dll is running. If the e-mail is from outside the company then the AD lookup will fail and probably will slow down msgreader

Greetings,
Kees

P.S. how much msg files did you import?
